streams/src/test/java/org/apache/kafka/streams/integration/AdjustStreamThreadCountTest.java

```diff
@@ -119,37 +157,44 @@ public void shouldAddStreamThread() throws Exception {
                     .sorted().toArray(),
                 equalTo(new String[] {"1", "2", "3"})
             );
-            waitForApplicationState(Collections.singletonList(kafkaStreams), KafkaStreams.State.REBALANCING, DEFAULT_DURATION);
-            waitForApplicationState(Collections.singletonList(kafkaStreams), KafkaStreams.State.RUNNING, DEFAULT_DURATION);
+
+            assertThat(waitForTransitionFromRebalancingToRunning(), is(true));
```

Comment:
It seems to me the method ```waitForTransitionFromRebalancingToRunning``` can do the assert as well because we always call ```assertThat(waitForTransitionFromRebalancingToRunning(), is(true)``` in this test.

Reply:
Good idea. Updated.